File: Cpp/Stacks.cpp
#include <iostream>
using namespace std;
#define Max 100
class Stack{
    private:
        int arr[Max];
        int top;
    public:
        Stack(){
            top = -1;
        }
        bool isempty(){
            return(top == -1);
        }
        bool isfull(){
            return(top == Max - 1);
        }
        void peek() {
            if(!isempty())
                cout << "Top element: " << arr[top] << endl;
            else
                cout << "Stack is empty\n";
        }

        void display() {
            if(!isempty()) {
                cout << "Stack elements: ";
                for(int i = 0; i <= top; i++)
                    cout << arr[i] << " ";
                cout << endl;
            } else {
                cout << "Stack is empty\n";
            }
        }
        void push(int val){
            if(!isfull()){
                top++;
                arr[top] = val;
            }
            else{
                cout<<"stack overflow !\n";
            }
        }

        void pop(){
            if(!isempty())
            {
                top--;
            }
            else
                cout<<"stack underflow\n";
        }

};

int main(){
    Stack s1;
    s1.push(5);
    s1.push(4);
    s1.push(3);
    s1.peek();
    s1.display();
    s1.pop();
    s1.display();
}

File: Python/stacks.py
class Stack:
    def __init__(self):
        self.stack = []

    def is_empty(self):
        return len(self.stack) == 0

    def push(self, val):
        self.stack.append(val)
        print(f"{val} pushed onto stack")

    def pop(self):
        if not self.is_empty():
            val = self.stack.pop()
            print(f"Popped: {val}")
            return val
        else:
            print("Stack Underflow!")
            return None

    def peek(self):
        if not self.is_empty():
            print(f"Top element: {self.stack[-1]}")
            return self.stack[-1]
        else:
            print("Stack is empty!")
            return None

    def display(self):
        if not self.is_empty():
            print("Stack elements:", self.stack)
        else:
            print("Stack is empty!")

s = Stack()
s.push(5)
s.push(4)
s.push(3)
s.peek()
s.display()
s.pop()
s.display()
